packages/browser-extension/src/hooks/useSteps.ts

```diff
@@ -82,114 +85,117 @@ const isExpectedDomElementState = async (
     return false;
   }
 
-  if (element === null) {
-    return step.assertion.require.notExist;
-  }
-  return step.assertion.require.exist;
-};
-
-const isStartPageStepReady = () => true;
-const isStartPageStepCompleted = wasUrlVisited;
-
-const isRedirectStepReady = () => true;
-const isRedirectStepCompleted = wasUrlVisited;
-
-const isUserActionStepReady = () => true;
-
-const isUserActionStepCompleted = isExpectedDomElementState;
+  const result = domStateAssertion(element, step.assertion);
+  storeAssertion(result);
 
-const isExpectUrlStepReady = () => true;
-const isExpectUrlStepCompleted = wasUrlVisited;
-
-const isNotarizeStepReady = isUrlRequestCompleted;
-const isNotarizeStepCompleted = hasProof;
-
-const isExtractVariablesStepReady = () => true;
-const isExtractVariablesStepCompleted = () => true;
-
-const isClickButtonStepReady = () => true;
-const isClickButtonStepCompleted = () => true;
-
-const checkStepCompletion: StepCompletions = {
-  startPage: isStartPageStepCompleted,
-  redirect: isRedirectStepCompleted,
-  userAction: isUserActionStepCompleted,
-  expectUrl: isExpectUrlStepCompleted,
-  notarize: isNotarizeStepCompleted,
-  extractVariables: isExtractVariablesStepCompleted,
-  clickButton: isClickButtonStepCompleted,
+  return result;
 };
 
-function checkCompletion<T extends ExtensionStep>(
-  step: T,
-): StepCompletionCheck<StepByType<T>> {
-  return checkStepCompletion[step];
+function intoActionableStep(
```

Comment:
This file looks quite too big imo, would you mind to move helper function to separate files?